src/callbox/renderer/CallboxApp.vue

```diff
@@ -0,0 +1,175 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+  -->
+
+<script setup lang="ts">
+import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import IconClose from 'vue-material-design-icons/Close.vue'
+import IconPhone from 'vue-material-design-icons/Phone.vue'
+import IconPhoneHangup from 'vue-material-design-icons/PhoneHangup.vue'
+import { t } from '@nextcloud/l10n'
+import { waitCurrentUserHasJoinedCall } from './callbox.service.ts'
+import { postBroadcast } from '../../shared/broadcast.service.ts'
+
+const AVATAR_SIZE = 15 * 1.5 * 2 // 2 lines
+const TIME_LIMIT = 45 * 1000
+
+const params = new URLSearchParams(window.location.search)
+
+const token = params.get('token')!
+const name = params.get('name')!
+const avatar = params.get('avatar')!
+
+/**
+ * Handle the call joined/missed outside the callbox
+ */
+waitCurrentUserHasJoinedCall(token, TIME_LIMIT).then((joined) => {
+	if (!joined) {
+		postBroadcast('notifications:missedCall', { name, token, avatar })
+	}
+	window.close()
+})
+
+/**
+ * Join the call
+ */
+async function join() {
+	postBroadcast('talk:conversation:open', { token, directCall: true })
+	window.close()
+}
+
+/**
+ * Dismiss the call
+ */
+function dismiss() {
+	window.close()
+}
+</script>
+
+<template>
+	<div class="callbox" @keydown.esc="dismiss">
```

Comment:
`@keydown.esc` doesn't work for me (guess that's because div is not a focused element, so `useHotKey()` should be used?)

Reply:
we should not have keyboard shortcuts if this window is automatically focused.
And if we have shortcuts they need to ignore the first second or something to make sure the user presses them intentionally in this window/dialog

Reply:
But we can address this in the future

Reply:
> It is not, as noted in the description and was discussed. Grabbing focus is not acceptable.

Ok cool, then proceed.